File: InverseKinematics.java
import java.util.Scanner; 

public class InverseKinematics
{
    //meters
    private static final double SHOULDER_LENGTH = 0.254; 
    private static final double ELBOW_LENGTH = 0.254; 


    static double x = 0; 
    static double y = 0; 
    static Scanner in = new Scanner( System.in ); 

    public static void main(String[] args)
    {
        System.out.println( "Enter: x:" );
        x = in.nextDouble(); 

        System.out.println( "Enter y:" );
        y = in.nextDouble(); 

        double elbowAngle = Math.acos( ( x * x + y * y - ( SHOULDER_LENGTH * SHOULDER_LENGTH ) - ( ELBOW_LENGTH * ELBOW_LENGTH ) ) / (2 * ELBOW_LENGTH * SHOULDER_LENGTH) );
        double shoulderAngle = Math.atan2( y, x ) - Math.atan2( (ELBOW_LENGTH * Math.sin(elbowAngle)), ( SHOULDER_LENGTH + (ELBOW_LENGTH * Math.cos(elbowAngle ) ) ) );

        elbowAngle = Math.toDegrees( elbowAngle ); 
        shoulderAngle = Math.toDegrees( shoulderAngle ); 

        shoulderAngle = 90 - shoulderAngle; 
        
        System.out.println( "Elbow Angle: " + elbowAngle );
        System.out.println( "Shoulder Angle: " + shoulderAngle );
    }
}

File: Inverse_Kinematics/InverseKinematics.py
import math

# meters
SHOULDER_LENGTH = 0.254
ELBOW_LENGTH = 0.254

x = 0
y = 0

print("Enter x:")
x = float(input())

print("Enter y:")
y = float(input())

elbow_angle = math.acos((x * x + y * y - SHOULDER_LENGTH * SHOULDER_LENGTH - ELBOW_LENGTH * ELBOW_LENGTH) / (2 * ELBOW_LENGTH * SHOULDER_LENGTH))
shoulder_angle = math.atan2(y, x) - math.atan2(ELBOW_LENGTH * math.sin(elbow_angle), SHOULDER_LENGTH + ELBOW_LENGTH * math.cos(elbow_angle))

elbow_angle = math.degrees(elbow_angle)
shoulder_angle = math.degrees(shoulder_angle)

shoulder_angle = 90 - shoulder_angle

print("Elbow Angle: ", elbow_angle)
print("Shoulder Angle: ", shoulder_angle)
